### DISCUSSION TITLE
Solver speed

### DISCUSSION THREAD

**Discussion by lecorrejm** [OpenSTREAM-solvers/openstream#24] [General]
The solver seems to be significantly slower than MEFISTO-T.
I think the main reason for this is due to object oriented programming with calculation of secondary parameters "on the fly" requiring doing the same things over and over.
The Matlab profiler identifies a couple of function that are particularly time-consuming.
We should have a look at this and decide if any optimization change could be made.

For instance, the three-field solver uses time-consuming methods of the mixture class (in particular Mixture.AFDISTR). However, all secondary parameters of the mixture class are fixed before the three-field solver is called.

**Comment by RWMorse** [comment]
We should discuss this today. 

Eliminating various uses of the annular flow distribution function has a significant impact on both the solution and the speed. I can cut the run time down about 25% just by removing two uses ofthe AFDISTR. But if I remove it from the wrong variable the solution may be wildly different.

**Comment by RWMorse** [comment]
[RunTime_RWM_22June2023.pdf](https://github.com/mfval/TwoPhaseSolver/files/11839442/RunTime_RWM_22June2023.pdf)

**Comment by jasonjan2003** [comment]
The worst offenders have been addressed in #26, #29. 

This discussion can be reopened later when new inefficiencies are found.
